p2p/src/peer_manager/mod.rs

```diff
@@ -1370,6 +1372,7 @@ where
             .get_or_create(peer, now, || {
                 self.peerdb
                     .known_addresses()
+                    .filter(|address| !self.peerdb.is_address_banned(&(*address).as_bannable()))
```

Comment:
I wasn't sure if this filter should be applied inside `known_addresses` function.

Reply:
I don't think this is implemented correctly anyway. The `is_address_banned` function doesn't take into account whether the banning has expired.

Reply:
A comment clarifying this in the code would be nice.

Reply:
I agree. It's difficult to figure this out based on a state.